### ISSUE TITLE
v2.21.0 Incorrect Cast Error for PostgreSQL Enum Type in Check Constraint

### ISSUE THREAD

**Issue by hc-12** [sbdchd/squawk#597]
**Squawk Version:** 2.21.0

**Summary:**  
Squawk incorrectly reports a syntax error for valid PostgreSQL enum type casts in check constraints using the `::` operator.

This appears to be a false positive in Squawk’s SQL parser for enum casts in check constraints.

**Migration SQL Example:**
```sql
alter table public.widget
  add constraint widget_config_schema_check check (
    checks.is_widget_config_valid('widget'::custom_types.widget_schema_type, config)
  ) not valid;

alter table public.widget_instance
  add constraint widget_instance_config_overrides_schema_check check (
    checks.is_widget_config_valid('widget_instance'::custom_types.widget_schema_type, config_overrides)
  ) not valid
```

### Squawk Output (paths omitted):

```
error[syntax-error]: Invalid cast. Use `cast`, `treat`, or `::` instead.
    |
    |     checks.is_widget_config_valid('widget'::custom_types.widget_schema_type, config)) not valid;
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
error[syntax-error]: Invalid cast. Use `cast`, `treat`, or `::` instead.
    |
    |     checks.is_widget_config_valid('widget_instance'::custom_types.widget_schema_type, config_overrides)
    |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```   

### Expected Behaviour:

PostgreSQL supports casting string literals to enum types using the `::` operator. The above syntax is valid and should not be flagged as an error.

### Actual Behaviour:

Squawk reports a syntax error and suggests using `::`, even though the code already uses it.

### Steps to Reproduce:

- Use a check constraint with a cast to a custom enum type as shown above.
- Run Squawk v2.21.0 against the migration file.

### Environment:

- PostgreSQL 15+
- Squawk v2.21.0

**Comment by sbdchd** [comment]
Oh oops, thank you for the report!

Fixed in latest:

```shell
❮ echo 'select call(x::int)' | npx squawk-cli@latest          
npm warn exec The following package was not found and will be installed: squawk-cli@2.21.1

Found 0 issues in 1 file 🎉

~/projects/squawk                                                                                                                                   steved-release-v2.21.1 22:17:28 3s310ms 
❯ echo 'select call(x::int)' | npx squawk-cli@2.21.0
npm warn exec The following package was not found and will be installed: squawk-cli@2.21.0
error[syntax-error]: Invalid cast. Use `cast`, `treat`, or `::` instead.
 --> stdin:1:13
  |
1 | select call(x::int)
  |             ^^^^^^
  |

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 1 issue in 1 file (checked 1 source file)
```
